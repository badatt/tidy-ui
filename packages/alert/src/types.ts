import { HTMLAttributes } from 'react';
import { TStatus } from '@tidy-ui/commons';
import { IBaseProps } from '@tidy-ui/types';

/**
 * Alert props
 *
 * @typedef {IAlertProps}
 * @interface IAlertProps
 * @augments {HTMLAttributes<HTMLDivElement>}
 * @public
 */
export interface IAlertProps extends IBaseProps, HTMLAttributes<HTMLDivElement> {
  /**
   * Element that encloses Alert
   */
  ele?: JSX.Element;

  /**
   * If `true`, the component filled with status tone
   *
   * @type {?true}
   * @default false
   */
  isFilled?: boolean;

  /**
   * If `true`, the component is created with sharp edges
   *
   * @type {?true}
   * @default false
   */
  isSharp?: boolean;

  /**
   * Color variant of the Alert, typically used to classify the
   * information based on its level
   *
   * @type {?TStatus}
   */
  status?: TStatus;
}

/**
 * AlertFooter props
 *
 * @typedef {IAlertFooterProps}
 * @interface IAlertFooterProps
 * @augments {HTMLAttributes<HTMLDivElement>}
 * @public
 */
export interface IAlertFooterProps extends IBaseProps, HTMLAttributes<HTMLDivElement> {
  /**
   * Element that encloses the footer
   */
  ele?: JSX.Element;
}

/**
 * AlertTitle props
 *
 * @typedef {IAlertTitleProps}
 * @interface IAlertTitleProps
 * @augments {HTMLAttributes<HTMLDivElement>}
 * @public
 */
export interface IAlertTitleProps extends IBaseProps, HTMLAttributes<HTMLDivElement> {
  /**
   * Element that encloses the title
   */
  ele?: JSX.Element;
}
