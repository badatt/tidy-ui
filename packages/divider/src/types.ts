import { HTMLAttributes } from 'react';
import { IBaseProps, TShade, TTone } from '@tidy-ui/types';

/**
 * Divider props
 */
export interface IDividerProps extends IBaseProps, HTMLAttributes<HTMLHRElement> {
  /**
   * Thickness of the divider
   *
   * @default 'thin'
   */
  density?: 'thin' | 'medium' | 'thick';
  /**
   * Top and bottom margins
   */
  margin?: string;
  /**
   * Shade of the color of the divider, 50 being low, 900 being high
   *
   * @type {?TShade}
   * @default 500
   */
  shade?: TShade;
  /**
   * The color of the divider
   *
   * @type {?TTone}
   * @default 'neutral'
   */
  tone?: TTone;

  /**
   * Divider variant
   *
   * @default 'solid'
   */
  variant?: 'dashed' | 'dotted' | 'double' | 'groove' | 'hidden' | 'ridge' | 'solid';

  /**
   * If `true`, set the divider vertically
   */
  vertical?: boolean;
}
