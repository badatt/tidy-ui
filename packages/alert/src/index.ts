export * from './Alert';
export * from './AlertFooter';
export * from './AlertTitle';
