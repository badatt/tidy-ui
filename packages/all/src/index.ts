export * from '@tidy-ui/commons';
export * from '@tidy-ui/feedback';
export * from '@tidy-ui/form';
export * from '@tidy-ui/layout';
export * from '@tidy-ui/navigation';
export * from '@tidy-ui/presentation';
