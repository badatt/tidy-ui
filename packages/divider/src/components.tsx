import { applyStandardOverrideStyles, css, styled } from '@tidy-ui/commons';
import { IDividerProps, IEnhancedDividerProps } from './types';

/** @internal */
const getShade = (isDark) => (isDark ? 700 : 300);

const DividerRoot = styled.hr<IDividerProps>`
  ${({ theme: { palette, isDark }, align, length, tone, shade, density, variant, margin, isVertical }) => css`
    border-color: ${palette[tone!][shade || getShade(isDark)]};
    flex-shrink: 0;
    border-style: ${variant};
    align-self: ${align};
    ${isVertical
      ? css`
          margin: 0 ${margin} !important;
          border-width: 0px ${density} 0px 0px;
          ${length &&
          css`
            height: ${length};
          `}
        `
      : css`
          margin: ${margin} 0 !important;
          border-width: 0px 0px ${density};
          ${length &&
          css`
            width: ${length};
          `}
        `}
  `}
  ${applyStandardOverrideStyles}
`;

const EnhancedDividerRoot = styled.div<IEnhancedDividerProps>`
  ${({ margin }) => css`
    display: flex;
    flex-direction: row;
    align-items: center;
    align-content: center;
    flex-wrap: nowrap;
    margin: ${margin} 0 !important;
  `}
  ${applyStandardOverrideStyles}
`;

const Separator = styled.hr<IEnhancedDividerProps>`
  ${({ theme: { palette, isDark }, tone, shade, density, variant }) => css`
    border-color: ${palette[tone!][shade || getShade(isDark)]};
    border-width: 0px 0px ${density};
    border-style: ${variant};
    flex-grow: 1;
  `}
`;

const Content = styled.div<IEnhancedDividerProps>`
  ${({ justify }) => css`
    ${justify === 'center' &&
    css`
      padding: 0 8px;
    `}
    ${justify !== 'center' &&
    (justify === 'start'
      ? css`
          padding: 0 8px 0 0;
        `
      : css`
          padding: 0 0 0 8px;
        `)}
  `}
`;

export { Content, DividerRoot, EnhancedDividerRoot, Separator };
