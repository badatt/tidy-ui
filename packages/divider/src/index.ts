export * from './Divider';
export { EnhancedDivider as DividerEnhanced } from './EnhancedDivider';
