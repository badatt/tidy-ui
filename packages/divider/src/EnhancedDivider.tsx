import React from 'react';
import { Content, EnhancedDividerRoot, Separator } from './components';
import { IEnhancedDividerProps } from './types';

/**
 * Enhanced Divider can be use to give a horizontal separation between components, a good replacement
 * for <hr> in html, with additional content support, powered by various styles (with theming) and settings
 */
const EnhancedDivider = React.forwardRef<HTMLHRElement, IEnhancedDividerProps>((props, ref) => {
  const { children, className, justify, ...rest } = props;
  const { tone, shade, density, variant } = rest;
  return (
    <EnhancedDividerRoot ref={ref} className={className} role="separator" {...rest}>
      {['center', 'end'].includes(justify!) && <Separator role="separator" {...{ density, shade, tone, variant }} />}
      {children && (
        <Content role="separator" justify={justify}>
          {children}
        </Content>
      )}
      {['center', 'start'].includes(justify!) && <Separator role="separator" {...{ density, shade, tone, variant }} />}
    </EnhancedDividerRoot>
  );
});

EnhancedDivider.defaultProps = {
  density: 'thin',
  justify: 'center',
  margin: '0.5rem',
  tone: 'neutral',
  variant: 'solid',
};

EnhancedDivider.displayName = 'EnhancedDivider';

export { EnhancedDivider };
