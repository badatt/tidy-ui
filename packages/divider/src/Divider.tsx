import React from 'react';
import { css, styled } from '@tidy-ui/commons';
import { EnhancedDivider } from './EnhancedDivider';
import { IDividerProps } from './types';

const DividerRoot = styled.hr<IDividerProps>`
  ${({ theme: { palette }, align, length, tone, shade, density, variant, margin, vertical }) => css`
    border-color: ${palette[tone!][shade!]};
    flex-shrink: 0;
    border-style: ${variant};
    align-self: ${align};
    ${vertical
      ? css`
          margin: 0 ${margin};
          border-width: 0px ${density} 0px 0px;
          ${length &&
          css`
            height: ${length};
          `}
        `
      : css`
          margin: ${margin} 0;
          border-width: 0px 0px ${density};
          ${length &&
          css`
            width: ${length};
          `}
        `}
  `}
`;

/** @internal */
interface DividerComponent
  extends React.ForwardRefExoticComponent<IDividerProps & React.RefAttributes<HTMLDivElement>> {
  /** @internal */
  Enhanced: typeof EnhancedDivider;
}

/**
 * Divider can be use to give a horizontal separation between components, a good replacement
 * for <hr> in html, powered by various styles (with theming) and settings
 */
const Divider = React.forwardRef<HTMLHRElement, IDividerProps>((props, ref) => {
  const { className, ...rest } = props;
  return <DividerRoot ref={ref} className={className} role="separator" {...rest} />;
}) as DividerComponent;

Divider.defaultProps = {
  density: 'thin',
  margin: '0.5rem',
  shade: 500,
  tone: 'neutral',
  variant: 'solid',
  vertical: false,
};

Divider.displayName = 'Divider';

Divider.Enhanced = EnhancedDivider;

export { Divider };
