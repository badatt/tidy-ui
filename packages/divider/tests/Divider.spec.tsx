/**
 * @jest-environment jsdom
 */
import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import '@testing-library/jest-dom';
import 'jest-styled-components';
import { orchidDark, orchidLight } from '@tidy-ui/commons';
import { Divider } from '../src';
import { Text } from '@tidy-ui/presentation';
import { TVariant } from '../src/types';

describe('Divider', () => {
  it('Basic render', () => {
    const tree = render(
      <ThemeProvider theme={orchidLight}>
        <Text>
          Lorem ipsum dolor sit amet consectetur adipisicing elit. Minima ratione, id vero quibusdam aperiam officia
          veritatis recusandae magni accusamus rem quo cupiditate architecto hic culpa eum totam reprehenderit, libero
          veniam!
        </Text>
        <Divider />
        <Text>
          Lorem ipsum dolor sit amet consectetur adipisicing elit. Minima ratione, id vero quibusdam aperiam officia
          veritatis recusandae magni accusamus rem quo cupiditate architecto hic culpa eum totam reprehenderit, libero
          veniam!
        </Text>
        <div style={{ display: 'flex' }}>
          <Text>
            Lorem ipsum dolor sit amet consectetur adipisicing elit. Quia sequi quam modi natus ipsum corrupti quis
            laudantium reiciendis error quas, voluptate eligendi possimus temporibus ipsam! Eveniet voluptates natus
            minima! Possimus.
          </Text>
          <Divider vertical />
          <Text>
            Lorem ipsum dolor sit amet consectetur adipisicing elit. Nobis accusantium ab magni assumenda, iure corporis
            natus voluptas laborum error corrupti inventore hic aliquid, debitis adipisci necessitatibus repellendus
            dolores ratione architecto.
          </Text>
        </div>
      </ThemeProvider>,
    );
    expect(tree).toMatchSnapshot();
  });
  it('Dark mode basic render', () => {
    const tree = render(
      <ThemeProvider theme={orchidDark}>
        <Text>
          Lorem ipsum dolor sit amet consectetur adipisicing elit. Minima ratione, id vero quibusdam aperiam officia
          veritatis recusandae magni accusamus rem quo cupiditate architecto hic culpa eum totam reprehenderit, libero
          veniam!
        </Text>
        <Divider />
        <Text>
          Lorem ipsum dolor sit amet consectetur adipisicing elit. Minima ratione, id vero quibusdam aperiam officia
          veritatis recusandae magni accusamus rem quo cupiditate architecto hic culpa eum totam reprehenderit, libero
          veniam!
        </Text>
      </ThemeProvider>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('Tone', () => {
    const tree = render(
      <ThemeProvider theme={orchidDark}>
        <Text>color major</Text>
        <Divider tone="major" />
      </ThemeProvider>,
    );
    expect(tree).toMatchSnapshot();
  });
  it('Density', () => {
    const tree = render(
      <ThemeProvider theme={orchidDark}>
        <Text>thick density</Text>
        <Divider density="thick" />
        <Text>thin density</Text>
        <Divider density="thin" />
        <Text>medium density</Text>
        <Divider density="medium" />
      </ThemeProvider>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('Shade', () => {
    const tree = render(
      <ThemeProvider theme={orchidDark}>
        <Text>800 shade</Text>
        <Divider shade={800} />
      </ThemeProvider>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('Variants', () => {
    const variants = ['dashed', 'dotted', 'double', 'groove', 'hidden', 'ridge', 'solid'];
    const tree = render(
      <ThemeProvider theme={orchidDark}>
        {variants.map((v, i) => (
          <div key={i}>
            <Text>{v} style</Text>
            <Divider variant={v as TVariant} />
            <Text>{v} style</Text>
          </div>
        ))}
      </ThemeProvider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
