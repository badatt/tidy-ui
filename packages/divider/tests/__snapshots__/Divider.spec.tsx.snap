// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider Basic render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c1 {
  border-color: hsla(0,0%,45%,1);
  margin: 0px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  border-width: 0px 0px thin;
  border-style: solid;
}

.c0 {
  display: block;
  margin: 0;
  font-size: 1rem;
  color: hsla(217,33%,17%,1);
  font-weight: 400;
  -webkit-letter-spacing: 0.00938em;
  -moz-letter-spacing: 0.00938em;
  -ms-letter-spacing: 0.00938em;
  letter-spacing: 0.00938em;
  line-height: 1.5;
}

<body>
    <div>
      <p
        class="c0"
        role="contentinfo"
      >
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Minima ratione, id vero quibusdam aperiam officia veritatis recusandae magni accusamus rem quo cupiditate architecto hic culpa eum totam reprehenderit, libero veniam!
      </p>
      <hr
        class="c1"
        role="separator"
      />
      <p
        class="c0"
        role="contentinfo"
      >
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Minima ratione, id vero quibusdam aperiam officia veritatis recusandae magni accusamus rem quo cupiditate architecto hic culpa eum totam reprehenderit, libero veniam!
      </p>
    </div>
  </body>,
  "container": .c1 {
  border-color: hsla(0,0%,45%,1);
  margin: 0px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  border-width: 0px 0px thin;
  border-style: solid;
}

.c0 {
  display: block;
  margin: 0;
  font-size: 1rem;
  color: hsla(217,33%,17%,1);
  font-weight: 400;
  -webkit-letter-spacing: 0.00938em;
  -moz-letter-spacing: 0.00938em;
  -ms-letter-spacing: 0.00938em;
  letter-spacing: 0.00938em;
  line-height: 1.5;
}

<div>
    <p
      class="c0"
      role="contentinfo"
    >
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Minima ratione, id vero quibusdam aperiam officia veritatis recusandae magni accusamus rem quo cupiditate architecto hic culpa eum totam reprehenderit, libero veniam!
    </p>
    <hr
      class="c1"
      role="separator"
    />
    <p
      class="c0"
      role="contentinfo"
    >
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Minima ratione, id vero quibusdam aperiam officia veritatis recusandae magni accusamus rem quo cupiditate architecto hic culpa eum totam reprehenderit, libero veniam!
    </p>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Divider Dark mode basic render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c1 {
  border-color: hsla(0,0%,45%,1);
  margin: 0px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  border-width: 0px 0px thin;
  border-style: solid;
}

.c0 {
  display: block;
  margin: 0;
  font-size: 1rem;
  color: hsla(210,40%,96%,1);
  font-weight: 400;
  -webkit-letter-spacing: 0.00938em;
  -moz-letter-spacing: 0.00938em;
  -ms-letter-spacing: 0.00938em;
  letter-spacing: 0.00938em;
  line-height: 1.5;
}

<body>
    <div>
      <p
        class="c0"
        role="contentinfo"
      >
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Minima ratione, id vero quibusdam aperiam officia veritatis recusandae magni accusamus rem quo cupiditate architecto hic culpa eum totam reprehenderit, libero veniam!
      </p>
      <hr
        class="c1"
        role="separator"
      />
      <p
        class="c0"
        role="contentinfo"
      >
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Minima ratione, id vero quibusdam aperiam officia veritatis recusandae magni accusamus rem quo cupiditate architecto hic culpa eum totam reprehenderit, libero veniam!
      </p>
    </div>
  </body>,
  "container": .c1 {
  border-color: hsla(0,0%,45%,1);
  margin: 0px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  border-width: 0px 0px thin;
  border-style: solid;
}

.c0 {
  display: block;
  margin: 0;
  font-size: 1rem;
  color: hsla(210,40%,96%,1);
  font-weight: 400;
  -webkit-letter-spacing: 0.00938em;
  -moz-letter-spacing: 0.00938em;
  -ms-letter-spacing: 0.00938em;
  letter-spacing: 0.00938em;
  line-height: 1.5;
}

<div>
    <p
      class="c0"
      role="contentinfo"
    >
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Minima ratione, id vero quibusdam aperiam officia veritatis recusandae magni accusamus rem quo cupiditate architecto hic culpa eum totam reprehenderit, libero veniam!
    </p>
    <hr
      class="c1"
      role="separator"
    />
    <p
      class="c0"
      role="contentinfo"
    >
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Minima ratione, id vero quibusdam aperiam officia veritatis recusandae magni accusamus rem quo cupiditate architecto hic culpa eum totam reprehenderit, libero veniam!
    </p>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Divider Other variants 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c1 {
  border-color: hsla(271,91%,65%,1);
  margin: 0px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  border-width: 0px 0px thin;
  border-style: solid;
}

.c2 {
  border-color: hsla(0,0%,45%,1);
  margin: 0px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  border-width: 0px 0px thick;
  border-style: solid;
}

.c3 {
  border-color: hsla(0,0%,15%,1);
  margin: 0px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  border-width: 0px 0px thin;
  border-style: solid;
}

.c4 {
  border-color: hsla(0,0%,45%,1);
  margin: 0px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  border-width: 0px 0px thin;
  border-style: dashed;
}

.c0 {
  display: block;
  margin: 0;
  font-size: 1rem;
  color: hsla(210,40%,96%,1);
  font-weight: 400;
  -webkit-letter-spacing: 0.00938em;
  -moz-letter-spacing: 0.00938em;
  -ms-letter-spacing: 0.00938em;
  letter-spacing: 0.00938em;
  line-height: 1.5;
}

<body>
    <div>
      <p
        class="c0"
        role="contentinfo"
      >
        color major
      </p>
      <hr
        class="c1"
        role="separator"
      />
      <p
        class="c0"
        role="contentinfo"
      >
        thick density
      </p>
      <hr
        class="c2"
        role="separator"
      />
      <p
        class="c0"
        role="contentinfo"
      >
        800 shade
      </p>
      <hr
        class="c3"
        role="separator"
      />
      <p
        class="c0"
        role="contentinfo"
      >
        dashed style
      </p>
      <hr
        class="c4"
        role="separator"
      />
    </div>
  </body>,
  "container": .c1 {
  border-color: hsla(271,91%,65%,1);
  margin: 0px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  border-width: 0px 0px thin;
  border-style: solid;
}

.c2 {
  border-color: hsla(0,0%,45%,1);
  margin: 0px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  border-width: 0px 0px thick;
  border-style: solid;
}

.c3 {
  border-color: hsla(0,0%,15%,1);
  margin: 0px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  border-width: 0px 0px thin;
  border-style: solid;
}

.c4 {
  border-color: hsla(0,0%,45%,1);
  margin: 0px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  border-width: 0px 0px thin;
  border-style: dashed;
}

.c0 {
  display: block;
  margin: 0;
  font-size: 1rem;
  color: hsla(210,40%,96%,1);
  font-weight: 400;
  -webkit-letter-spacing: 0.00938em;
  -moz-letter-spacing: 0.00938em;
  -ms-letter-spacing: 0.00938em;
  letter-spacing: 0.00938em;
  line-height: 1.5;
}

<div>
    <p
      class="c0"
      role="contentinfo"
    >
      color major
    </p>
    <hr
      class="c1"
      role="separator"
    />
    <p
      class="c0"
      role="contentinfo"
    >
      thick density
    </p>
    <hr
      class="c2"
      role="separator"
    />
    <p
      class="c0"
      role="contentinfo"
    >
      800 shade
    </p>
    <hr
      class="c3"
      role="separator"
    />
    <p
      class="c0"
      role="contentinfo"
    >
      dashed style
    </p>
    <hr
      class="c4"
      role="separator"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
